cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/profile/ProfileDataset.java

```diff
@@ -199,19 +218,82 @@ private void changeProfileStatus(ProfileId profileId,
                                     expectedStatus, oldProfile.getProvisioner()));
   }
 
-  private MDSKey getRowKey(NamespaceId namespaceId) {
-    return getRowKey(namespaceId, null);
+  /**
+   * Get assignments with the profile.
+   *
+   * @param profileId the profile id
+   * @return the set of entities that the profile is assigned to
+   * @throws NotFoundException if the profile is not found
+   */
+  public Set<EntityId> getProfileAssignments(ProfileId profileId) throws NotFoundException {
+    Profile profile = table.get(getProfileRowKey(profileId), Profile.class);
+    if (profile == null) {
+      throw new NotFoundException(profileId);
+    }
+
+    return Collections.unmodifiableSet(new HashSet<>(table.list(getEntityIndexRowKeyPrefix(profileId),
+                                                                EntityId.class)));
+  }
+
+  /**
+   * Add an assignment to the profile. Assignment can only be added if the profile is ENABLED
+   *
+   * @param profileId the profile id
+   * @param entityId the entity to add to the assgiment
+   */
+  public void addProfileAssignment(ProfileId profileId,
+                                   EntityId entityId) throws ProfileConflictException, NotFoundException {
+    Profile profile = table.get(getProfileRowKey(profileId), Profile.class);
+    if (profile == null) {
+      throw new NotFoundException(profileId);
+    }
+    if (profile.getStatus() == ProfileStatus.DISABLED) {
+      throw new ProfileConflictException(
+        String.format("Profile %s is DISABLED. So no entity can be assigned to it", profileId.getProfile()), profileId);
```

Comment:
It's better not to start sentences with 'So'. 

is DISABLED. No entity can be assigned to it.